src/Server/Repository/TokenRepositoryInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrOAuth2\Server\Repository;
+
+use ZfrOAuth2\Server\Entity\AbstractToken;
+
+/**
+ * Interface TokenRepositoryInterface
+ */
+interface TokenRepositoryInterface
+{
+    /**
+     * @param AbstractToken $token
+     * @return AbstractToken
+     */
+    public function save(AbstractToken $token): AbstractToken;
+
+    /**
+     * @param string $token
+     * @return AbstractToken|null
+     */
+    public function findByToken(string $token);
+
+    /**
+     * @return AbstractToken[]
+     */
+    public function findAll(): array;
+
+    /**
+     * @param AbstractToken $token
+     * @return null
+     */
+    public function deleteToken(AbstractToken $token): null;
```

Comment:
:null is not a valid return type as far as I know. Should simply @return void (and no return typehint as typehint on void will only be added in PHP 7.1)